### PR TITLE
feat: align consumer configs

### PR DESCRIPTION
**Pull request by ludusrusso**
This PR allignes consumer config type with nats jeststream configurations.

/cc @wallyqs

### CODE REVIEW

**Comment by ludusrusso** [comment]
I'm not able to locally run tests. I've to run them as super user to make it work. Any help on this?

```
❯ make test                          
go vet ./controllers/... ./pkg/natsreloader/...
go test -race -cover -count=1 -timeout 10s ./controllers/... ./pkg/natsreloader/...
ok      github.com/nats-io/nack/controllers/jetstream   5.562s  coverage: 44.2% of statements
?       github.com/nats-io/nack/pkg/natsreloader        [no test files]
--- FAIL: TestReloader (0.01s)
    natsreloader_test.go:113: error adding "/private/var/folders/fs/hp0g8wk95pb98hf0zlbc1cdm0000gn/T" for "/var/folders/fs/hp0g8wk95pb98hf0zlbc1cdm0000gn/T": operation not permitted
FAIL
coverage: [no statements]
FAIL    github.com/nats-io/nack/pkg/natsreloader/natsreloadertest       0.097s
FAIL
make: *** [test] Error 1
```

File: controllers/jetstream/consumer.go [hunk 1]
```diff
@@ -312,6 +314,39 @@ func createConsumer(ctx context.Context, c jsmClient, spec apis.ConsumerSpec) (e
...
+			if err != nil {
+				return err
+			}
+			bo = append(bo, d)
+		}
+		opts = append(opts, func(o *jsmapi.ConsumerConfig) error {
+			o.BackOff = bo
+			return nil
+		})
+	}
```
Comment (ludusrusso) [inline controllers/jetstream/consumer.go:348]: Not super sure about this!

File: go.mod [hunk 2]
```diff
@@ -5,8 +5,8 @@ go 1.16
 require (
 	github.com/fsnotify/fsnotify v1.5.0
 	github.com/google/gofuzz v1.2.0 // indirect
-	github.com/nats-io/jsm.go v0.0.26
-	github.com/nats-io/nats.go v1.12.1
+	github.com/nats-io/jsm.go v0.0.31
+	github.com/nats-io/nats.go v1.14.1-0.20220412004736-c75dfd54b52c
```
Comment (ludusrusso) [inline go.mod:9]: Updating jsm to latest versions force `nats.go` version to `v1.14.1-0.20220412004736-c75dfd54b52c`. 
Should I use v0.0.30 or it's ok?

File: pkg/jetstream/apis/jetstream/v1beta2/consumertypes.go [hunk 3]
```diff
@@ -45,6 +45,11 @@ type ConsumerSpec struct {
 	StreamName        string   `json:"streamName"`
 	TLS               TLS      `json:"tls"`
 	Account           string   `json:"account"`
+	HeadersOnly       bool     `json:"headers_only"`
+	MaxRequestBatch   int      `json:"max_batch"`
+	MaxRequestExpires string   `json:"max_expires"`
+	MaxWaiting        int      `json:"max_waiting"`
+	BackOff           []string `json:"backoff"`
```
Comment (ludusrusso) [inline pkg/jetstream/apis/jetstream/v1beta2/consumertypes.go:52]: `BackOff` is of time `[]time.Duration`. It is ok to map it in kube with `[]string`?

**Comment by ludusrusso** [comment]
I think I need to update also `deploy/crds.ymal`. It should be autogenerated right?

**Comment by wallyqs** [comment]
Thanks @ludusrusso, the makefile will generate some of the code with `make jetstream-controller` but I think the `deploy/crds.yml` is not autogenerated and has to be crafted by hand for the new fields. /cc @variadico

File: go.mod (hunk 2 above)
Comment (wallyqs) [reply go.mod:9]: It is fine to use the pseudo version one

Comment (ludusrusso) [reply go.mod:9]: thanks

File: deploy/crds.yml [hunk 4]
```diff
@@ -513,6 +513,26 @@ spec:
...
+                default: '' 
+              maxWaiting:
+                description: number of outstanding pulls that are allowed on any one consumer.  Pulls made that exceeds this limit are discarded.
+                type: integer
+              backoff:
+                description: series of intervals by which retries will be attempted for this consumr
+                type: array
+                items:
+                  type: string
+                default: []
```
Comment (ludusrusso) [inline deploy/crds.yml:535]: Also update crds.yaml. Not sure about description and default values :D

**Comment by wallyqs** [comment]
Included as part of #78

**Pull request closed without merging**